apps/web/scripts/integrity-hashes.cjs

```diff
@@ -4,14 +4,109 @@ const crypto = require('crypto')
+    if (entry.isDirectory()) {
+      results = results.concat(getAllChunkFiles(entryPath))
+    } else if (entry.isFile() && entry.name.endsWith('.js')) {
+      results.push(entryPath)
+    }
+  }
+  return results
+}
+
+/**
+ * Compute the SHA-384 SRI hash for a given file
+ */
+function computeSriHash(filePath) {
+  const content = fs.readFileSync(filePath)
+  const hash = crypto.createHash('sha384').update(content).digest('base64')
+  return `sha384-${hash}`
+}
+
+/**
+ * Build a mapping from each chunk's public path to its integrity hash
+ * e.g.: { "/_next/static/chunks/foo.js": "sha384-abc..." }
+ */
+function buildSriManifest() {
+  const allJsFiles = getAllChunkFiles(CHUNKS_DIR)
+  const manifest = {}
+
+  for (const filePath of allJsFiles) {
+    // filePath is absolute, e.g. /path/to/out/_next/static/chunks/foo.js
+    // We want to create a key like "/_next/static/chunks/foo.js"
+    const relPath = path.relative(OUT_DIR, filePath).replace(/\\/g, '/')
+    // On Windows, ensure forward slashes
+    const publicPath = `/${relPath}`
+    manifest[publicPath] = computeSriHash(filePath)
+  }
+
+  return manifest
+}
+
+/**
+ * Write the manifest file in `out/_next/static/`.
+ */
+function writeExternalManifest(manifestObj) {
+  // We'll store it in out/_next/static/chunks-sri-manifest.js or somewhere similar
+  const manifestJson = JSON.stringify(manifestObj, null, 2)
+
+  // The script sets the global window.__CHUNK_SRI_MANIFEST
+  const fileContents = `
+/**
+ * Auto-generated chunk SRI manifest.
+ * DO NOT EDIT.
+ */
+window.__CHUNK_SRI_MANIFEST = ${manifestJson};
+`
+
+  // Place it in out/_next/static/ for a consistent path
+  const manifestJsPath = path.join(OUT_DIR, '_next', 'static', MANIFEST_JS_FILENAME)
+  fs.writeFileSync(manifestJsPath, fileContents, 'utf8')
+
+  // Return the public path so we can insert <script src="..."> references
+  return `/_next/static/${MANIFEST_JS_FILENAME}`
+}
+
+/**
+ * Insert a single <script src="..."> reference into each .html file
+ */
+function insertManifestScriptIntoHtml(manifestScriptPath) {
+  function processDir(dir) {
+    const entries = fs.readdirSync(dir, { withFileTypes: true })
+    for (const entry of entries) {
+      const entryPath = path.join(dir, entry.name)
+      if (entry.isDirectory()) {
+        processDir(entryPath)
+      } else if (entry.isFile() && entry.name.endsWith('.html')) {
+        // Load the HTML, inject one <script src="..." >
+        const html = fs.readFileSync(entryPath, 'utf8')
+        const $ = cheerio.load(html)
+
+        // Ideally, put it in <head> so it loads early
+        // so the manifest is available by the time Next tries dynamic chunks
+        if ($('head').length) {
+          $('head').append(`\n<script src="${manifestScriptPath}"></script>\n`)
+        } else {
+          // Fallback: if no <head>, put it at start of <body>
+          $('body').prepend(`\n<script src="${manifestScriptPath}"></script>\n`)
```

Comment:
Not sure this is a good idea. This script itself lacks a hash.

Alternatively, we could store the manifest file as JSON, fetch and cache the manifest in runtime.

```js
        // Set integrity if we have a matching hash
        const hashManifest = window. __CHUNK_SRI_MANIFEST || await fetch('/out/sri-manifest.json').then(r => r.json())
        if (hashManifest[src]) {
            script.integrity = hashManifest[src];
        }
```

Reply:
I was following the implementation of the `__buildManifest.js` that Next generates and imports in the same way. After we add the script to the document we run `addSRIToAllHtmlFiles` which hashes all the static imports including the manifest script
<img width="1152" alt="Screenshot 2025-03-06 at 09 17 33" src="https://github.com/user-attachments/assets/7babfa0a-353f-4c7e-a371-f5992b8084a8" />